robot/markers.py

```diff
@@ -46,6 +46,45 @@ def rot_y_degrees(self) -> Degrees:
         return Degrees(math.degrees(self.rot_y_radians))
 
 
+_Orientation = NamedTuple('Orientation', (
+    ('rot_x_radians', Radians),
+    ('rot_y_radians', Radians),
+    ('rot_z_radians', Radians),
+))
+
+
+class Orientation(_Orientation):
+    """
+    Represents the orientation in 3d space as rotations around x, y, and z axes.
```

Comment:
What is the order of application of the angles, and/or do they always apply to the cartesian axes of the camera?

A camera-relative interpretation of `(45°, 90°, 0)` is as follows: consider a marker which is leaning forwards by 45° but placed on a horizontal rotating platform (like a "lazy susan"); then rotate the platform by 90°.

A marker-relative interpretation is as follows: consider a marker perpendicular to a rotating platform which is angled 45° towards the observer (like a lazy susan on the surface of a table which has been half-pushed over); then rotate the platform by 90°.

In both scenarios the marker is now facing off to the side, however in the former its baseline is still horizontal while in the latter its baseline is at a 45° to the horizontal.

I'm not sure which of the two systems is easier to think about, nor which we've got data for. We should find out and document this however.

Note that if we apply the angles in the reverse order (`z` then `y` then `x`) then the two scenarios above change roles.

Reply:
@Adimote I see this is partially resolved by 6018425c576e63f34063727ab17e0d0b9c786e31, though we still don't describe whether the rotations are relative to the marker or the camera.

I _think_ I hope that they're relative to the fixed frame of the camera (I'm expecting that that would make the angles for a pair of markers more easily comparable), though I've not explore enough examples to know which is better/easier to think about.

Reply:
Took me a while to figure this out, but I'm pretty sure the rotations are actually applied relative to the marker. I've added this detail in 7535af1202c33c19a20484e90e379cde0c357f4d This means an upside-down marker would report inverse `rot_y`. We might want to investigate how to make this easier for the students to use in a later update.